app/javascript/modules/listings/DirectoryHelpers.tsx

```diff
@@ -246,6 +248,16 @@ export const openListingsView = (
   listings.length > 0 &&
   getListingCards(listings, directoryType, stackedDataFxn, filtersSet, useUpdatedDirectoryStatuses)
 
+export const fcfsSalesView = (
```

Comment:
**question (non-blocking):**  So currently this is the same view as openListings. I am assuming we will want to make some visual changes for fcfs listings?

Reply:
no idts - it is supposed to be in the same view, so that the listings are all the lottery listings and then all the fcfs listings in one view

Reply:
**note:** The only difference between fcfs sales and lottery listings will be the image card status but that is handled in `getListingCards`. Should we reuse `openListingsView`?

Reply:
I'm cool with that, if that is clear as a reader.

Reply:
My main reason for creating a new function was readability, since fcfsListings are logically distinct from open in how we sort them.

Reply:
Makes sense! I can go either way, Im ok with keeping the function for readability.

Reply:
Maybe call it openListingsWithFcfs though? bc it's not actually just fcfs listings

Reply:
the listings.fcfsSales should only contain fcfsListings. They display below listings.open. Here is how they are separated out before render:
```
// Sort listings in four buckets based on their status and filters
export const sortListings = (
  listings: RailsListing[],
  filters: EligibilityFilters,
  setMatch: React.Dispatch<React.SetStateAction<boolean>>
) => {
  const open = []
  const upcoming = []
  const results = []
  const additional = []
  const fcfsSales = []
  listings.forEach((listing) => {
    if (isFcfsSalesListing(listing)) {
      fcfsSales.push(listing)
    } else if (dayjs(listing.Application_Due_Date) > dayjs()) {
      if (!filters || listing.Does_Match) {
        setMatch(true)
        open.push(listing)
      } else {
        additional.push(listing)
      }
    } else {
      if (isLotteryComplete(listing)) {
        results.push(listing)
      } else {
        upcoming.push(listing)
      }
    }
  })
 ```

Then we just pass them into the wrapper function and render right below the open listings
```
 {fcfsSalesView(
                listings.fcfsSales,
                props.directoryType,
                props.getSummaryTable,
                hasFiltersSet,
                isSalesFcfsEnabled
              )}
```

Reply:
ah, gotcha!